### PR TITLE
CORE refactoring, added Symfony/Console, bin/build improvement, improvement in test suite

### PR DESCRIPTION
**Pull request by reisraff**

### CODE REVIEW

**Comment by mauri870** [comment]
Good work @reisraff :+1:

File: .gitignore [hunk 1]
```diff
@@ -1,9 +1,28 @@
-material/
-vendor/
+# root
 cache.properties
-bin/phpcs
+
+# bin
+bin/pdepend
```
Comment (henriquemoody) [inline .gitignore]:
```
bin/*
!bin/php-gameboy
```

File: boot.php [hunk 2]
```diff
@@ -1,60 +1,7 @@
 <?php
 
+require_once('vendor/autoload.php');
```
Comment (henriquemoody) [inline boot.php:17]:
``` php
require_once __DIR__.'/vendor/autoload.php'; 
```

Comment (henriquemoody) [inline boot.php:17]: I'm sorry, actually you must keep the previous autoloader

Comment (mauri870) [inline boot.php:17]: Yep, this require don't work on global installation with composer

**Comment by henriquemoody** [comment]
I think we can go back to review that after #30 is merged, there are too much things out of the scope of _added Symfony/Console_ here.

---

I think one pull request must have only one goal, but that's just my opinion.

**Comment by gabrielrcouto** [comment]
Really good work @reisraff, but sorry, we will go without symfony/console. What you prefer: discard this PR or just keep the build improvements?

**Comment by reisraff** [comment]
@gabrielrcouto discard this PR I'll open other with just build improvements.

**Pull request closed without merging**